x-pack/packages/kbn-cloud-security-posture-common/utils/helpers.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { QueryDslQueryContainer } from '@kbn/data-views-plugin/common/types';
+import type { CspBenchmarkRulesStates } from '../schema/rules/latest';
+
+export const extractErrorMessage = (e: unknown, defaultMessage = 'Unknown Error'): string => {
+  if (e instanceof Error) return e.message;
+  if (typeof e === 'string') return e;
+
+  return defaultMessage; // TODO: i18n
+};
+
+export const buildMutedRulesFilter = (
+  rulesStates: CspBenchmarkRulesStates
+): QueryDslQueryContainer[] => {
+  const mutedRules = Object.fromEntries(
+    Object.entries(rulesStates).filter(([key, value]) => value.muted === true)
+  );
+
+  const mutedRulesFilterQuery = Object.keys(mutedRules).map((key) => {
+    const rule = mutedRules[key];
+    return {
+      bool: {
+        must: [
+          { term: { 'rule.benchmark.id': rule.benchmark_id } },
+          { term: { 'rule.benchmark.version': rule.benchmark_version } },
+          { term: { 'rule.benchmark.rule_number': rule.rule_number } },
+        ],
+      },
+    };
+  });
+
+  return mutedRulesFilterQuery;
+};
```

Comment:
can we also add a test for this function? it requires some thinking to understand the format of the input object and the output array